### PR TITLE
BUILD: print test name on failure in builds.sh

### PR DESCRIPTION
**Pull request by michal-shalev**
## What
Added logic to `builds.sh` to print the test name if a test fails.

## Why ?
To quickly identify which test failed, improving debugging efficiency.

## How ?
Inserted a conditional check after each test in builds.sh to print the test name and exit on failure.

### CODE REVIEW

File: buildlib/tools/builds.sh [hunk 1]
```diff
@@ -461,5 +461,5 @@ do
 	[ -d "${ucx_build_dir}" ] && rm -rf ${ucx_build_dir}/*
 
 	# run the test
-	${test_name}
+	${test_name} || { echo "Test failed: ${test_name}"; exit 1; }
```
Comment (tvegas1) [inline buildlib/tools/builds.sh:464]: if we exit, we cannot get a list of failing tests at once, which could be painful?

**Review by tvegas1 — COMMENTED** [review]
maybe don't

File: buildlib/tools/builds.sh (hunk 1 above)
Comment (michal-shalev) [reply buildlib/tools/builds.sh:464]: I can remove the "exit 1;" part

Comment (yosefe) [reply buildlib/tools/builds.sh:464]: the test would already exit after first failure, to not consume CI resources
https://github.com/openucx/ucx/blob/master/buildlib/tools/builds.sh#L1

Comment (tvegas1) [reply buildlib/tools/builds.sh:464]: I see

Comment (michal-shalev) [reply buildlib/tools/builds.sh]: removed the "exit 1;" part @tvegas1 @yosefe

Comment (yosefe) [reply buildlib/tools/builds.sh]: pls keep the exit

Comment (michal-shalev) [reply buildlib/tools/builds.sh:464]: done @yosefe

**Review by tvegas1 — DISMISSED** [review]

**Review by yosefe — DISMISSED** [review]

File: buildlib/tools/builds.sh (hunk 1 above)
Comment (gleon99) [inline buildlib/tools/builds.sh:464]: Maybe better use `azure_log` helpers here instead, @yosefe  wdyt?

Comment (michal-shalev) [reply buildlib/tools/builds.sh]: Changed `echo` to `azure_log_error`, please check @gleon99 @yosefe

**Review by gleon99 — APPROVED** [review]

**Comment by gleon99** [comment]
Please squash

**Review by yosefe — APPROVED** [review]

**Pull request merged**